File: Week1/pset1/cash/cash.c
#import <stdio.h>
#import <cs50.h>
#import <math.h>

int get_coin_count(float change_due);

int main(void)
{
    //promt user for a positive float value
    float change_due;
    do
    {
        change_due = get_float("Give us your change:\n");
    }
    while (change_due <= 0.0);

    int coin_count = get_coin_count(change_due);//get coin_count using our created function
    printf("%i\n", coin_count);

}

int get_coin_count(float change_due)
{
    //get change in int to avoid rounding errors
    int change_due_int = (int) round(change_due * 100);

    int coin_count = 0; //Keep track of the number of coins we will give back
    //find out how many quarters
    int quarter = 25;
    while (change_due_int >= quarter)
    {
        coin_count++;
        change_due_int -= quarter;
    }
    int dime = 10;
    while (change_due_int >= dime)
    {
        coin_count++;
        change_due_int -= dime;
    }
    int nickle = 5;
    while (change_due_int >= nickle)
    {
        coin_count++;
        change_due_int -= nickle;
    }
    int penny = 1;
    while (change_due_int >= penny)
    {
        coin_count++;
        change_due_int -= penny;
    }
    return coin_count;
}

File: pset6/cash/cash.py
from cs50 import get_float
"""Take a float input and return the fewest numbet of coins needed to make change
EX: If 0.51 is needed the result would be 3. 2 quarters (0.25) and one penny (0.01)"""

# promt user for positive float
while True:
    change = get_float("Change owed: ")
    if change > 0.0:
        break

# Keep track of coins
coins_due = 0

# Change change to int
change = change * 100

if change >= 25:
    quarters = int(change / 25)
    change = change - (quarters * 25)
    coins_due += quarters

if change >= 10:
    dimes = int(change / 10)
    print(dimes)
    change = change - (dimes * 10)
    coins_due += dimes
    print(change)
if change >= 5:
    nickels = int(change / 5)
    change = change - (nickels * 5)
    coins_due += nickels

if change > 0:
    pennies = int(change / 1)
    change = change - (pennies * 1)
    coins_due += pennies

print(coins_due)
